core/src/main/scala/org/apache/spark/ui/jobs/UIData.scala

```diff
@@ -180,8 +180,8 @@ private[spark] object UIData {
         speculative = taskInfo.speculative
       )
       newTaskInfo.gettingResultTime = taskInfo.gettingResultTime
-      newTaskInfo.setAccumulables(taskInfo.accumulables.filter {
-        accum => !accum.internal && accum.metadata != Some(AccumulatorContext.SQL_ACCUM_IDENTIFIER)
+      newTaskInfo.setAccumulables(taskInfo.accumulables.filter { acc =>
+        !acc.internal && !acc.metadata.contains(AccumulatorContext.SQL_ACCUM_IDENTIFIER)
```

Comment:
We should revert this one too for the same reason in https://github.com/apache/spark/pull/17458/files#r108432518